src/CheckoutSdk/Sessions/Source/SessionSource.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Checkout.Common;
+using Newtonsoft.Json;
+
+namespace Checkout.Sessions.Source
+{
+    public abstract class SessionSource
+    {
+        [JsonProperty(PropertyName = "type")]
+        protected readonly SessionSourceType Type;
+
+        [JsonProperty(PropertyName = "billing_address")]
+        protected SessionAddress _billingAddress;
+
+        [JsonProperty(PropertyName = "home_phone")]
+        protected Phone _homePhone;
+
+        [JsonProperty(PropertyName = "mobile_phone")]
+        protected Phone _mobilePhone;
+
+        [JsonProperty(PropertyName = "work_phone")]
+        protected Phone _workPhone;
+
+        protected internal SessionSource(SessionSourceType type, SessionAddress billingAddress, Phone homePhone,
```

Comment:
Why this method is internal, and not protected.
Same question about constructors.

Reply:
I have removed the internal modifier.